File: services/frontend/src/lib/api/clients.ts
// Client directory API — talks to the gateway's /api/v1/clients endpoint.
// One row per ad-client (Noise / NOI today). The dashboards list, the New
// Dashboard form, and any future client-scoped UI all read from here so the
// frontend's client list is the same one the database holds.

import { apiRequest } from './http';
import { gatewayBase } from '@/lib/api/gateway';

const BASE_URL = gatewayBase();

export interface ClientRecord {
  id: string;
  slug: string;
  name: string;
  initials: string;
  accent_color: string;
  logo_path: string | null;
}

export const clientsApi = {
  /** Every client. Small set; no pagination needed. */
  async list(): Promise<ClientRecord[]> {
    const data = await apiRequest<{ clients: ClientRecord[] }>(
      `${BASE_URL}/api/v1/clients`,
    );
    return data.clients ?? [];
  },
};

File: services/backend/gateway/api/clients.py
"""HTTP API for the platform's client directory.

One row per ad-client. Drives the New Dashboard form's client picker and
the dashboard card's brand badge. Today there is one client (Noise / NOI);
future clients are seeded via direct INSERT — there is intentionally no
client-creation UI yet (admins add clients, the app consumes them).
"""

from __future__ import annotations

from fastapi import APIRouter, Depends

from .auth import CurrentUser, current_user
from .db import get_pool

router = APIRouter(prefix="/clients", tags=["clients"])


@router.get("")
async def list_clients(
    _user: CurrentUser = Depends(current_user),
):
    """Every client, oldest first. Small set; no pagination needed today."""
    pool = await get_pool()
    rows = await pool.fetch(
        """
        SELECT id::text AS id, slug, name, initials, accent_color, logo_path
        FROM clients
        ORDER BY created_at ASC
        """
    )
    return {"clients": [dict(r) for r in rows]}
